diff --git a/lua/presence/file_explorers.lua b/lua/presence/file_explorers.lua
index 050d8e7..242615d 100644
--- a/lua/presence/file_explorers.lua
+++ b/lua/presence/file_explorers.lua
@@ -5,6 +5,7 @@ return {
     ["[defx] default-"] = "Defx",
     ["netrw"] = "Netrw",
     ["TelescopePrompt"] = "Telescope",
-    ['neo-tree'] = 'Neotree',
-    ['fern'] = 'Fern'
+    ["neo-tree"] = "Neotree",
+    ["fern"] = "Fern",
+    ["oil"] = "Oil",
 }
diff --git a/lua/presence/init.lua b/lua/presence/init.lua
index 1d2aefe..b9a0fb7 100644
--- a/lua/presence/init.lua
+++ b/lua/presence/init.lua
@@ -430,7 +430,7 @@ function Presence:get_project_name(file_path)
     local project_path = vim.fn.system(project_path_cmd)
     project_path = vim.trim(project_path)
 
-    if project_path:find("fatal.*") then
+    if project_path:find("fatal.*") or project_path == os.getenv("HOME") then
         self.log:info("Not a git repository, skipping...")
         return nil
     end
@@ -756,6 +756,10 @@ function Presence:get_buttons(buffer, parent_dirpath)
             repo_url = string.format("https://%s", relative)
         end
 
+        if repo_url:match("^https://github%.com/.+/dotfiles%.git$") and not parent_dirpath:match("/%.bash%.d") then
+            return nil
+        end
+
         self.log:debug(string.format("Adding button with repository URL: %s", repo_url))
 
         return {
